contracts/otx-lock/src/validate/sighash_all_acp.rs

```diff
@@ -0,0 +1,91 @@
+use crate::helper::{add_prefix, new_blake2b, verify_pubkey_hash};
+use crate::types::SIGHASH_ALL_SIGNATURE_SIZE;
+use crate::{error::Error, types::SighashMode};
+
+use ckb_lib_secp256k1::LibSecp256k1;
+
+// Import heap related library from `alloc`
+// https://doc.rust-lang.org/alloc/index.html
+use alloc::{vec, vec::Vec};
+
+// Import CKB syscalls and structures
+// https://docs.rs/ckb-std/
+use ckb_std::{
+    ckb_constants::Source,
+    ckb_types::bytes::Bytes,
+    ckb_types::packed::WitnessArgsBuilder,
+    ckb_types::prelude::*,
+    error::SysError,
+    high_level::{load_transaction, load_witness_args},
+};
+
+pub(crate) fn validate_sighash_all_anyonecanpay(
+    lib: &LibSecp256k1,
+    index: usize,
+    signature: &[u8; SIGHASH_ALL_SIGNATURE_SIZE],
+    expected_pubkey_hash: &[u8],
+) -> Result<(), Error> {
+    let tx = load_transaction()?.raw();
+
+    // input
+    let input = tx.inputs().get(index).ok_or(Error::Encoding)?;
+    let input_len = input.as_slice().len() as u64;
+
+    // outputs
+    let outputs = tx.outputs();
+    let outputs_count = outputs.len();
+    let outputs_len = outputs.as_slice().len() as u64;
+
+    // witness
```

Comment:
Leave a comment here: Sighash mode does not cover witnesses at positions beyond the number of inputs.